ui/src/timeMachine/components/variableToolbar/VariableItem.tsx

```diff
@@ -1,25 +1,42 @@
 // Libraries
-import React, {PureComponent} from 'react'
+import React, {FunctionComponent, useRef, useState} from 'react'
+
+// Components
+import VariableTooltipContents from 'src/timeMachine/components/variableToolbar/VariableTooltipContents'
+import BoxTooltip from 'src/shared/components/BoxTooltip'
 
 // Types
 import {Variable} from '@influxdata/influx'
 
-// Styles
-import 'src/timeMachine/components/fluxFunctionsToolbar/FluxFunctionsToolbar.scss'
-
 interface Props {
   variable: Variable
 }
 
-class VariableItem extends PureComponent<Props> {
-  public render() {
-    const {variable} = this.props
-    return (
-      <dl className="variables-toolbar--item">
-        <dd className="variables-toolbar--label">{variable.name}</dd>
-      </dl>
-    )
+const VariableItem: FunctionComponent<Props> = ({variable}) => {
+  const trigger = useRef<HTMLDivElement>(null)
+  const [tooltipVisible, setTooltipVisible] = useState(false)
+
+  let triggerRect: DOMRect = null
+
+  if (trigger.current) {
+    triggerRect = trigger.current.getBoundingClientRect() as DOMRect
   }
+
+  return (
+    <div
+      className="variables-toolbar--item"
+      onMouseEnter={() => setTooltipVisible(true)}
```

Comment:
nit. these anonymous functions will be recreated every time the component renders.

Reply:
I was thinking that's OK in this case, since they aren't being passed to a child component that uses them in deciding whether to render or not.

https://reactjs.org/docs/hooks-faq.html#are-hooks-slow-because-of-creating-functions-in-render

Reply:
oh, nice!